code/game/objects/effects/misc.dm

```diff
@@ -111,6 +111,8 @@
 
 /obj/effect/forcefield/fog/CanPass(atom/movable/mover, turf/target)
 	if(isxeno(mover))
+		var/mob/living/carbon/xenomorph/X = mover
+		X.empty_gut()
```

Comment:
This should be in /Crossed, behavior shouldn't happen in a Can function

Reply:
I agree but /Crossed would be too late.

Reply:
If it happens in `Crossed()` the marine will be expelled over the fog tile.
In here it will happen before, which is what's intended.